build.gradle

```diff
@@ -193,6 +193,24 @@ project(':iceberg-hive') {
   }
 }
 
+project(':iceberg-mr') {
+  dependencies {
+    compile project(':iceberg-api')
+    compile project(':iceberg-core')
+    compile project(':iceberg-hive')
+    compile project(':iceberg-orc')
+    compile project(':iceberg-parquet')
+
+    compileOnly("org.apache.hadoop:hadoop-client") {
+      exclude group: 'org.apache.avro', module: 'avro'
+    }
+
+    testCompile project(path: ':iceberg-hive', configuration: 'testArtifacts')
```

Comment:
Why do we need to depend on `iceberg-hive` module? Not sure if this will potentially introduce circular dependency.

Reply:
This is so that I can rely on `HiveCatalog`. Yes you are right though, we'd have to figure out how to arrange out modules when we have a Hive storage handler

Reply:
Maybe this could just rely on a catalog from the API, and include iceberg-hive for tests. That wouldn't introduce a dependency cycle.

Reply:
We'd need the `org.apache.iceberg.hive.HiveCatalogs#loadCatalog` which is available in iceberg-hive

Reply:
I think we have some options. We could add a config option to the builder to set the catalog, or we could use reflection to call `HiveCatalogs.loadCatalog` if the class is present.

Reply:
I can see a future where we have Hive SerDes and StorageHandlers in the `hive` subproject which in turn will need to depend on the input format(s) here so we need to avoid a circular dependency right? In our [InputFormat](https://github.com/ExpediaGroup/hiveberg/blob/master/src/main/java/com/expediagroup/hiveberg/IcebergInputFormat.java) up to now we've just use HadoopTables and the JobConf to get everything we need.

Reply:
I fixed this. For now, I've adding iceberg-hive as a test dep

Reply:
Maybe iceberg-hive as a test dep can also cause recursive deps. If that's the case we can simply test with `HadoopCatalog` instead and remove iceberg-hive altogether

Reply:
I've removed iceberg-hive altogether. Relying on HadoopCatalog to test the Catalog plugin